services/passport.js

```diff
@@ -41,7 +41,7 @@ passport.serializeUser((ID, done) => {
 passport.deserializeUser((ID, done) => {
     dbService.getUserByID(ID, function(result) {
         // Convert "dbUser" (RowDataPacket in JSON form) to User
-        var user = new User(result[0]);
+        var user = new User(result);
         done(null, user);                    
```

Comment:
Since this query returns ONE row at most, we should return "result[0]" within the query and NOT every single parent caller.